src/uct/ib/rc/accel/rc_mlx5_ep.c

```diff
@@ -653,6 +648,19 @@ ucs_status_t uct_rc_mlx5_ep_flush(uct_ep_h tl_ep, unsigned flags,
         sn = ep->tx.wq.sig_pi;
     }
 
+    if (ucs_unlikely(flags & UCT_FLUSH_FLAG_CANCEL)) {
+        ucs_assert(!(ep->super.flags & UCT_RC_EP_FLAG_FLUSH_CANCEL));
+        uct_ep_pending_purge(&ep->super.super.super, NULL, 0);
```

Comment:
no need purge pending - UCP should do it
UCT would return NO_RESOURCE if has pending (can assert here)